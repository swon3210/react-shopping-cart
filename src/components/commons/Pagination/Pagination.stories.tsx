import Pagination, { Props } from './Pagination';

export default {
  title: 'Components/Commons/Pagination',
  component: Pagination,
};

const Template = (args: Props) => <Pagination {...args}></Pagination>;

export const Default = Template.bind({});
(Default as any).args = {};
