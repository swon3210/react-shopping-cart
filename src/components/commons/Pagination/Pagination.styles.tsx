import styled from 'styled-components';

export const Pagination = styled.div`
  display: inline-flex;
  justify-content: space-between;
  align-items: center;
`;

export const PaginationButtonWrapper = styled.div``;

export const PaginationButton = styled.button<{ isActive: boolean }>`
  display: inline-flex;
  justify-content: center;
  align-items: center;
  width: 35px;
  height: 35px;
  font-size: 18px;
  line-height: 21px;
  border-radius: 4px;
  ${({ theme, isActive }) => isActive && `background-color: ${theme.PRIMARY_COLOR}`};
  color: ${({ theme, isActive }) => (isActive ? theme.WHITE : theme.TEXT_COLOR)};
  transition: color 0.5s, background-color 0.5s;

  &:not(:last-child) {
    margin-right: 10px;
  }

  &:hover {
    color: ${({ theme }) => theme.WHITE};
    background-color: ${({ theme }) => theme.MINT_500};
  }
`;

export const PrevButtonWrapper = styled.button`
  margin-right: 24px;
`;

export const NextButtonWrapper = styled.button`
  margin-left: 24px;
`;
