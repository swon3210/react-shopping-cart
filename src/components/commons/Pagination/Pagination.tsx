import * as Styled from './Pagination.styles';

export interface Props {
  maxPageNumber?: number;
  onPaginationButtonClick: () => void;
  onPrevButtonClick: () => void;
  onNextButtonClick: () => void;
}

const Pagination = ({ maxPageNumber = 10, onPaginationButtonClick, onPrevButtonClick, onNextButtonClick }: Props) => {
  const PaginationButtonList = [...Array(maxPageNumber)].map((_, index) => (
    <Styled.PaginationButton>{index + 1}</Styled.PaginationButton>
  ));

  return (
    <Styled.Pagination>
      <Styled.PrevButtonWrapper onClick={onPrevButtonClick}>{'<'}</Styled.PrevButtonWrapper>
      <Styled.PaginationButtonWrapper onClick={onPaginationButtonClick}>
        {PaginationButtonList}
      </Styled.PaginationButtonWrapper>
      <Styled.NextButtonWrapper onClick={onNextButtonClick}>{'>'}</Styled.NextButtonWrapper>
    </Styled.Pagination>
  );
};

export default Pagination;
