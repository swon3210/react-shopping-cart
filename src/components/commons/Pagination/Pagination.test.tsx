import Pagination from './Pagination';
import { render } from '@testing-library/react';

describe('Pagination Component', () => {
  it('Pagination Snapshot', () => {
    const onNextButtonClick = () => {
      console.log('onNextButtonClick');
    };

    const onPaginationButtonClick = () => {
      console.log('onPaginationButtonClick');
    };

    const onPrevButtonClick = () => {
      console.log('onPrevButtonClick');
    };

    const pageTitleUtil = render(
      <Pagination
        onNextButtonClick={onNextButtonClick}
        onPaginationButtonClick={onPaginationButtonClick}
        onPrevButtonClick={onPrevButtonClick}
      />
    );
    expect(pageTitleUtil.asFragment()).toMatchSnapshot();
  });
});
