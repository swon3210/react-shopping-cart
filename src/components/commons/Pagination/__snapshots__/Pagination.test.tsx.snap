// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component Pagination Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa bYwOME"
  >
    <button
      class="sc-hKwCoD jFpMmS"
    >
      &lt;
    </button>
    <div
      class="sc-gsDJrp"
    >
      <button
        class="sc-dkPtyc fYKpJs"
      >
        1
      </button>
      <button
        class="sc-dkPtyc fYKpJs"
      >
        2
      </button>
      <button
        class="sc-dkPtyc fYKpJs"
      >
        3
      </button>
      <button
        class="sc-dkPtyc fYKpJs"
      >
        4
      </button>
      <button
        class="sc-dkPtyc fYKpJs"
      >
        5
      </button>
      <button
        class="sc-dkPtyc fYKpJs"
      >
        6
      </button>
      <button
        class="sc-dkPtyc fYKpJs"
      >
        7
      </button>
      <button
        class="sc-dkPtyc fYKpJs"
      >
        8
      </button>
      <button
        class="sc-dkPtyc fYKpJs"
      >
        9
      </button>
      <button
        class="sc-dkPtyc fYKpJs"
      >
        10
      </button>
    </div>
    <button
      class="sc-eCImvq bRJVsM"
    >
      &gt;
    </button>
  </div>
</DocumentFragment>
`;
